core/Model/CalendarModel.vala

```diff
@@ -294,6 +290,36 @@ public class Maya.Model.CalendarModel : Object {
 
     //--- Helper Methods ---//
 
+    /** Set the week_starts_on property: the first day of the week.
+     *
+     * Locale handling is based on information from
+     * https://sourceware.org/glibc/wiki/Locales
+     */
+    private GLib.DateWeekday get_week_start () {
+        // Set the "baseline" for start of week: Sunday or Monday?
+        uint week_day1 = (uint) Posix.NLTime.WEEK_1STDAY.to_string ();
+        var week_1stday = 0; // Default to 0 if unrecognized data
+        if (week_day1 == 19971130) { // Sunday
+            week_1stday = 0;
+        } else if (week_day1 == 19971201) { // Monday
+            week_1stday = 1;
+        } else {
+            warning ("Unknown value of _NL_TIME_WEEK_1STDAY: %u", week_day1);
+        }
+
+        // Get the start of week
+        int week_start_posix = Posix.NLTime.FIRST_WEEKDAY.to_string ().data[0];
```

Comment:
Again using `int.parse ()` makes more sense to me and less reliant on implementation details.

Reply:
I agree, but I don't think I can do anything else. What the call to langinfo actually produces is a `  union { unsigned int word; char *string; }`, which I don't know how to deal with otherwise in Vala. (I don't know how to deal with unions in For reference, here's what GNOME Calendar does, which can interface with its C weirdness directly:
```c
  union { unsigned int word; char *string; } langinfo;
  gint week_1stday = 0;
  gint first_weekday = 1;
  guint week_origin;

  langinfo.string = nl_langinfo (_NL_TIME_FIRST_WEEKDAY);
  first_weekday = langinfo.string[0]; // My line 311: get the first byte of the string
  langinfo.string = nl_langinfo (_NL_TIME_WEEK_1STDAY);
  week_origin = langinfo.word; // My line 300: get the word.
```

When I call `to_string ()`, it produces the string `\u0001\u0002\u0001`, which afaik is just a sequence of 3 bytes. `int.parse` doesn't work on this string and just returns 0. So I don't believe that using a parser could work. The use of `.data` just gets the first byte, which is what the C implementations are doing. (Why first byte? I have no idea, just following everyone else)

Maybe there's a better way to do this. I haven't actually found any documentation for what on earth `nl_langinfo` actually returns, so it's really tough to reason about it.

Reply:
OK, its obviously more complicated than it at first appears from the Vala syntax so I may have to take the view "if it works it must be OK", unless a more knowledgeable dev wants to jump in.